app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala

```diff
@@ -123,29 +123,15 @@ class HelpToSaveConnectorImpl @Inject() (http: WSHttp)(implicit ec: ExecutionCon
 
   private val eligibilityURL = s"$helpToSaveUrl/help-to-save/eligibility-check"
 
-  private def toEligibilityCheckResponse(eligibilityCheckResponse: EligibilityCheckResponse): Either[String, EligibilityCheckResult] = {
-    val (reasonCode, reason) = eligibilityCheckResponse.reasonCode → eligibilityCheckResponse.reason
-
-    eligibilityCheckResponse.resultCode match {
-      case 1 ⇒
-        Either.fromOption(
-          EligibilityReason.fromInt(reasonCode, reason).map(EligibilityCheckResult.Eligible),
-          s"Could not parse ineligibility reasonCode '$reasonCode'. Reason string was '$reason'")
-
-      case 2 ⇒
-        Either.fromOption(
-          IneligibilityReason.fromInt(reasonCode, reason).map(EligibilityCheckResult.Ineligible),
-          s"Could not parse eligibility reasonCode '$reasonCode'. Reason string was '$reason'")
-
-      case 3 ⇒
-        Right(EligibilityCheckResult.AlreadyHasAccount(reason))
-
-      case other ⇒
-        Left(s"Could not parse eligibility result code '$other'. Result string was '$reason'")
-
+  // scalastyle:off magic.number
+  private def validate(response: EligibilityCheckResponse): Either[String, EligibilityCheckResponse] = {
+    (response.resultCode, response.reasonCode) match {
+      case (1, 6) | (1, 7) | (1, 8)          ⇒ Right(response) // eligible
+      case (2, 2) | (2, 3) | (2, 4) | (2, 5) ⇒ Right(response) // non-eligible
+      case (3, 1)                            ⇒ Right(response) // account already opened
+      case invalid                           ⇒ Left(s"Could not parse eligibility response, unknown combination (resultCode, reasonCode) = ${(invalid._1, invalid._2)}")
```

Comment:
I'm just wondering here if we need to check the 2nd number, as this increases validation coverage, but also increases fragility.  The whole idea of changing Account Already Created to 3,1 was to just use the resultCode for our three-way outcome.